src/index.js

```diff
@@ -1,37 +1,36 @@
 function log(prevState, action, nextState) {
-  console.group('%c action', 'color: gray; font-weight: lighter;', action.name);
-  console.log('%c prev state', 'color: #9E9E9E; font-weight: bold;', prevState);
-  console.log('%c data', 'color: #03A9F4; font-weight: bold;', action.data);
-  console.log('%c next state', 'color: #4CAF50; font-weight: bold;', nextState);
-  console.groupEnd();
+  console.group("%c action", "color: gray; font-weight: lighter;", action.name)
+  console.log("%c prev state", "color: #9E9E9E; font-weight: bold;", prevState)
+  console.log("%c data", "color: #03A9F4; font-weight: bold;", action.data)
+  console.log("%c next state", "color: #4CAF50; font-weight: bold;", nextState)
+  console.groupEnd()
 }
 
-export default function(options) {
-  options = options || {};
-  options.log = typeof options.log === 'function' ? options.log : log;
-
-  return function(emit) {
-    var actionStack = [];
-    return {
-      events: {
-        action: function(state, actions, action) {
-          actionStack.push(action);
-        },
-        resolve: function(state, actions, result) {
-          if (typeof result === 'function') {
-            var action = actionStack.pop();
-            return function(update) {
-              return result(function(result) {
-                actionStack.push(action);
-                return update(result);
-              });
-            };
-          }
-        },
-        update: function(state, actions, nextState) {
-          options.log(state, actionStack.pop(), nextState);
+function onStateUpdate(callback) {
+  return function() {
+    return function(action) {
+      return function(result) {
+        return function(update) {
+          return update(function(prevState) {
+            if (typeof result === "function") {
+              return result(function(updateResult) {
+                callback(prevState, action, updateResult)
+                return updateResult
```

Comment:
@okwolf Doesn't...
```js
return callback(prevState, action, updateResult)
```
...work as well?

Reply:
Wouldn't that require `callback` to return the the `updateResult`/`nextState`? Even doing that it still breaks the state updates in my tests for some reason 🤔

Reply:
I must be forgetting something or something is not making sense. What does `callback` return then?

Reply:
Currently `undefined` since it's only used for logging the action and state values without modifying them.

Reply:
@okwolf And wht if you returned `updateResult` from that function?

Reply:
Correction, it does work if you remember to always return the new state from the callback function, including your custom `log`. I wasn't intending this to be a state validation helper though...

Reply:
Under that assumption we could rewrite this as:
```js
function onStateUpdate(callback) {
  return function() {
    return function(action) {
      return function(result) {
        return function(update) {
          return update(function(prevState) {
            if (typeof result === "function") {
              return result(function(updateResult) {
                return callback(prevState, action, updateResult)              
              })
            } else {
              return callback(prevState, action, result)
            }
          })
        }
      }
    }
  }
}
```

Reply:
@okwolf Okay, and how would that change usage?

Reply:
@JorgeBucaran any custom `log` you pass would come with this gotcha:
```js
hooks: [
  logger({
    log(prevState, action, nextState) {
      // log stuff...
      return nextState;
    }
  })
]
```

Reply:
@okwolf Not if you wrap the user's log function.

Reply:
True, but then don't I basically have the code I had before where I ignore the result of `log`?

Reply:
You do get the user's log function no? I guess you could wrap that function so that it behaves in a way that lets you write everything like  in https://github.com/hyperapp/logger/pull/6#discussion_r141998519

Reply:
@JorgeBucaran yes, the `log` function is currently the only option available. The old `events.update` would only override the state update [if you returned `false`](https://github.com/hyperapp/hyperapp/blob/master/docs/events.md#update). I'm not sure what the best API is for the new `hooks` approach to this.